rfcs/text/0000-dss-deletion-daemon.md

```diff
@@ -0,0 +1,268 @@
+file. And instead of physically removing a file version, the DCP hides that
+file version behind a deletion marker.
+
+Under certain circumstances however, the *physical deletion* of a file version
+is required. The first section of this specification describes the
+circumstances and mechanisms for logical and physical deletion of files and the
+bundles that contain them.
+
+### User Stories
+
+*Share the [User Stories](https://www.mountaingoatsoftware.com/agile/user-stories) motivating this RFC.*
+As an operator of the DSS, I found unconsented data in the DSS that must be removed, to protect user's privacy.
+
+As a user of the DSS, I would like to know what bundles have been deleted, so I can keep my personal index up-to-date. 
+
+## Scientific "guardrails" [optional]
+
+*Describe recommended or mandated review from HCA Science governance to ensure that the RFC addresses the needs of the scientific community.*
+
+## Detailed Design
+
+*Explain the design in sufficient detail such that the implementation and (if appropriate) the interaction with existing DCP software are both reasonably clear.*
+
+### Reasons
+
+Reasons for deletion of (meta)data are
+
+* consent to share (meta)data derived from a donor's specimens is withdrawn by
+  that donor after the (meta)data was submitted to DCP
+  
+* that consent was never given but this fact is revealed only after the
+  (meta)data was submitted to the DCP
+  
+* storing or processing the (meta)data causes a serious disruption of service
+  within the DCP
+
+* the (meta)data was submitted to the DCP by a malicious party in violation of
+  the DCP's data sharing agreement. This includes (meta)data that violates the
+  laws or regulations in a jurisdiction applicable to the DCP or the (meta)data
+  it stores
+
+Whether any of the reasons require a physical or logical deletion is at the
+discretion of the administrator performing the deletion. That discretion is
+guided by a standard operating procedure (SOP).
+
+### Process
+
+Once it is determined that a (meta)data file needs to be physically deleted –
+the details of this determination are the scope of an SOP – the following
+procedure takes place:
+
+1) An administrator identifies the Data Store file versions to be deleted. The
+   administrator then identifies the Data Store bundle versions referencing
+   those file versions.
+
+2) The administrator runs the admin CLI tool to queue the bundle for deletion using the bundles UUID and version.
+   The bundle UUID and version is required for deletion. The admin CLI will return a list of files that will be 
+   deleted and bundles that are effected by this deletion. Bundles may be affected if they share files or are derived 
+   from the deleted bundle.
+   
+   ```bash
+   delete_bundle.py <uuid>.<version> --reason [consent_withdrawn, consent_absent, service_disruption, legal] 
+    --type [logical, physical] --details "<additional info>"
```

Comment:
Is this meant to be an exhaustive list of possible reasons? I ask because I don't think the DSS should impose any constraints on the reason for deletion. New reasons may arise.